nxdrive/engine/engine.py

```diff
@@ -713,12 +711,10 @@ def cancel_upload(self, transfer_uid: int) -> None:
 
     def cancel_session(self, uid: int) -> None:
         """Cancel all transfers for given session."""
-        session_uploads = self.dao.get_session_uploads(uid)
-        if not session_uploads:
-            return
-        for upload in session_uploads:
-            self.cancel_upload(upload["uid"])
         self.dao.change_session_status(uid, TransferStatus.CANCELLED)
+        batchs_to_cancel = self.dao.cancel_session(uid)
+        for batch in batchs_to_cancel:
+            self.remote.cancel_batch(batch)
```

Comment:
```suggestion
        self.dao.cancel_session(uid)
        # We could cancel all batches, but in reality it would freeze the GUI.
        # Cancelling a session with a lot of items is worthless the use of a specific thread
        # to do the clean-up ourselves. Let the server doing it for us. We will be able to
        # tackle a possible issue in time.
        # for batch in self.dao.cancel_session(uid):
        #     self.remote.cancel_batch(batch)
```